lib/matplotlib/tests/test_dates.py

```diff
@@ -511,6 +511,68 @@ def _create_auto_date_locator(date1, date2):
         assert strings == expected
 
 
+def test_concise_formatter_tz():
+    def _create_auto_date_locator(date1, date2, tz):
+        fig, ax = plt.subplots()
+
+        locator = mdates.AutoDateLocator(interval_multiples=True)
+        formatter = mdates.ConciseDateFormatter(locator, tz=tz)
+        ax.yaxis.set_major_locator(locator)
+        ax.yaxis.set_major_formatter(formatter)
+        ax.set_ylim(date1, date2)
+        fig.canvas.draw()
+        sts = []
+        for st in ax.get_yticklabels():
+            sts += [st.get_text()]
+        return sts, ax.yaxis.get_offset_text().get_text()
+
+    d1 = datetime.datetime(1997, 1, 1).replace(tzinfo=datetime.timezone.utc)
+    results = ([datetime.timedelta(weeks=52 * 200),
+                [str(t) for t in range(1980, 2201, 20)],
+                ""
+                ],
+               [datetime.timedelta(weeks=52),
+                ['1997', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug',
+                'Sep', 'Oct', 'Nov', 'Dec'],
+                "1997"
+                ],
+               [datetime.timedelta(days=141),
+                ['Jan', '22', 'Feb', '22', 'Mar', '22', 'Apr', '22',
+                 'May', '22'],
+                "1997-May"
+                ],
+               [datetime.timedelta(days=40),
+                ['Jan', '05', '09', '13', '17', '21', '25', '29', 'Feb',
```

Comment:
should we just remove the tests where the TZ doesn't matter?  

Does the locator need the tz as well?  I'd have thought the labels would be the same, but the tick locations would change.  This is locating the ticks in GMT and then labelling them as the TZ....

Reply:
> should we just remove the tests where the TZ doesn't matter?

Sure

> Does the locator need the tz as well? I'd have thought the labels would be the same, but the tick locations would change. This is locating the ticks in GMT and then labeling them as the TZ....

To the best of my knowledge:

The locator does not care about the timezone of the data its dealing with because it works with the number of minutes since the epoch, which is independent of any timezone that any input datestamp might have, and any timezone you might want to display this information as. But given a number of minutes since the epoch, the formatter's job is to print it and therefore does require the timezone.

Reply:
Thats fair enough, but we *could* specify the timezone we would like the locator to work in to agree w/ the formatter.  But not for this PR obviously....  

I don't use timezones so input from users who do is very valuable...

Reply:
BTW, thanks a lot for getting the tests to work, appreciate that its a bit of work the first time!

Reply:
NP, glad to help.

Reply:
OK, so the locator does take a TZ argument.  So its possible to do everything in Pacific time if that is what you want...  Not that this should change this PR...